### PR TITLE
[WFS / OAPIF] Emit request headers of initial request when following a redirect

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: src/providers/wfs/qgsbasenetworkrequest.cpp [hunk 1]
```diff
@@ -434,8 +437,11 @@ bool QgsBaseNetworkRequest::sendPOSTOrPUTOrPATCH( const QUrl &url, const QByteAr
     logMessageIfEnabled();
     return false;
   }
-  request.setHeader( QNetworkRequest::ContentTypeHeader, contentTypeHeader );
-  for ( const QNetworkReply::RawHeaderPair &headerPair : extraHeaders )
+
+  mRequestHeaders = extraHeaders;
+  mRequestHeaders <<  QNetworkReply::RawHeaderPair( "Content-Type", contentTypeHeader.toUtf8() );
+
+  for ( const QNetworkReply::RawHeaderPair &headerPair : mRequestHeaders )
```
Comment (m-kuhn) [inline src/providers/wfs/qgsbasenetworkrequest.cpp:444]:
```suggestion
  for ( const QNetworkReply::RawHeaderPair &headerPair : std::as_const( mRequestHeaders ) )
```

File: src/providers/wfs/qgsbasenetworkrequest.cpp [hunk 2]
```diff
@@ -634,6 +640,9 @@ void QgsBaseNetworkRequest::replyFinished()
           request.setAttribute( QNetworkRequest::CacheLoadControlAttribute, mForceRefresh ? QNetworkRequest::AlwaysNetwork : QNetworkRequest::PreferCache );
           request.setAttribute( QNetworkRequest::CacheSaveControlAttribute, true );
 
+          for ( const QNetworkReply::RawHeaderPair &headerPair : mRequestHeaders )
```
Comment (m-kuhn) [inline src/providers/wfs/qgsbasenetworkrequest.cpp:643]:
```suggestion
          for ( const QNetworkReply::RawHeaderPair &headerPair : std::as_const( mRequestHeaders ) )
```

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
worth backporting?

File: src/providers/wfs/qgsbasenetworkrequest.cpp (hunk 1 above)
Comment (rouault) [reply src/providers/wfs/qgsbasenetworkrequest.cpp]: done

File: src/providers/wfs/qgsbasenetworkrequest.cpp (hunk 2 above)
Comment (rouault) [reply src/providers/wfs/qgsbasenetworkrequest.cpp]: done

**Comment by rouault** [comment]
> worth backporting?

done in https://github.com/qgis/QGIS/pull/53014

**Pull request merged**